Mathlib/Data/Set/Card.lean

```diff
@@ -1055,5 +1056,132 @@ theorem ncard_eq_three : s.ncard = 3 ↔ ∃ x y z, x ≠ y ∧ x ≠ z ∧ y 
   · rwa [ENat.coe_toNat] at h; rintro h'; simp [h'] at h
   simp [h]
 
+theorem exists_union_disjoint_cardinal_eq_of_even_finite [DecidableEq α] (s : Set α)
```

Comment:
It looks like this one is a lot easier to prove first for finsets:
```
theorem exists_disjoint_union_of_even_card [DecidableEq α] (s : Finset α) (he : Even #s) :
    ∃ (t u : Finset α), t ∪ u = s ∧ Disjoint t u ∧ #t = #u :=
  let ⟨n, hn⟩ := he
  let ⟨t, ht, ht'⟩ := exists_subset_card_eq (show n ≤ #s by omega)
  ⟨t, s \ t, by simp [card_sdiff, disjoint_sdiff, *]⟩
```
and then you can transfer that to a result for `Set`, which should make this proof quite a bit nicer!

Reply:
That is nice! Fixed:)